### DISCUSSION TITLE
Quarto appears to change body-footer entry slightly

### DISCUSSION THREAD

**Discussion by bryanhanson** [quarto-dev/quarto-cli#1058] [Q&A]
I am using [Goatcounter](https://goatcounter.com) for web page analytics.  I just figured out after a month that it is not working.  Goatcounter says to put this on the page so it will be tracked:

```
<script data-goatcounter="https://chemospec.goatcounter.com/count" async src="//gc.zgo.at/count.js"></script>
```

So I put in my `_quarto.yml` the following:

```
website:
  body-footer: <script data-goatcounter="https://chemospec.goatcounter.com/count" async src="//gc.zgo.at/count.js"></script>
```

But when the `body-footer` entry is actually put on the page Quarto changes it just a bit:

```
<script data-goatcounter="https://chemospec.goatcounter.com/count" async="" src="../..//gc.zgo.at/count.js"></script>
```

Any suggestions for doing this differently?  Or is there a bug in how `body-footer` is being parsed?

Full repo is [here](https://github.com/bryanhanson/CSR)  Last updated around 3 May using then-current Quarto. However, I don't see any discussion of `body-footer` in either issues or discussions that I can find since that time.

**Comment by cscheid** [comment]
I haven't attempted a full repro, but I suspect there are two potential issues here, one of which looks like a straight quarto bug (or at least misfeature):

1. Pandoc appears to be converting `async` to `async=""`, which _might_ behave badly. MDN [describes the behavior of `async`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script) as "... if the async attribute is present", which to me implies thta this should _not_ cause any problems. But I don't know if web browsers will interpret `async=""` as `async="false"` or something like that.

2. Quarto is interpreting paths starting with `//` as (what we call) a "root-relative path", and translating them to a path that resolves to a location as if `/gc.zgo.at/count.js` existed as a file in the project. This seems to me to be a clear quarto bug (or misfeature). `//` means "use the same scheme as the current request, but go to a different server". The spec [is here](https://www.rfc-editor.org/rfc/rfc1808#section-4). This exists so that requests behave correctly under both HTTP and HTTPS. Clearly we're misbehaving here. With that said, you should be able to work around it by explicitly writing `https://` or `http://`, depending on how your file will be served.

Thanks for the report! I'll file a bug for the second point, but file handling is a finicky part of quarto and we're hoping not to make big changes between now and 1.0.

(cc @jjallaire)

**Reply by jjallaire** [reply]
@cscheid your diagnosis (and suggested workaround) are right on. I think this will be an easy fix for us (we have a function that checks for external URLs that just needs to be made a bit smarter)

**Reply by cscheid** [reply]
Cool. I should be able to fix that later today.

**Reply by bryanhanson** [reply]
You all are the best!  Thank you.

**Comment by jjallaire** [comment]
Should be resolved here: https://github.com/quarto-dev/quarto-cli/commit/fb6a2a41e1c59cfb811efc38add379605dccdb7d

**Comment by bryanhanson** [comment]
Looks right (with async=“”), I will report w/i a day or two if the counter is working.

Thank you!

**Comment by bryanhanson** [comment]
The Goatcounter is working correctly.  Thanks so much!
